### ISSUE TITLE
[Bug]: intercepting dump() output does not work after update

### ISSUE THREAD

**Issue by blumewas** [beyondcode/herd-community#409]
### Operating system version

macOS Sonoma 14.1.1

### System architecture

ARM64 (M1, M2, etc)

### Herd Version

1.4.1 (Build: 21)

### PHP Version

PHP 8.2.16 (cli) (built: Feb 16 2024 00:30:02) (NTS)

### Bug description

After the upgrade from Herd 1.4.0 to 1.4.1 the dump interception does not work properly anymore.

When using `dump()` or `dd()` the output seems not to be intercepted by Herd. The application states that the server is running and the port was not adjusted. Sometimes executing `herd restart` inside the console reenables the feature. But after a system reboot the same error occurs.

A simple reinstall of the application has not worked either.

If anymore information is required I would be glad to help.

**Comment by sschlein** [comment]
I know this might be obvious, but have you tried disabling and enabling dump interception? I can't come up with a reason why it shouldn't work if the server is running.

Can you also please check your related php.ini file and see if there is the line:
```
auto_prepend_file=/Users/seb/Library/Developer/Xcode/DerivedData/PHP-axtuyydbmljqpnbbhmxsoluabffo/Build/Products/Debug/Herd.app/Contents/Resources/valet/dump-loader.php
```

![CleanShot 2024-02-17 at 12 44 44](https://github.com/beyondcode/herd-community/assets/2911113/da560ee0-d446-4fa6-bb10-2e9469ed7bd3)

**Comment by blumewas** [comment]
Hey,
thanks for the quick reply. Inside the php.ini that shows up when using Herd's show php.ini function I can find at line 6:
```
auto_prepend_file=/Applications/Herd.app/Contents/Resources/valet/dump-loader.php
```

Toggling the interception as described works while I've tested it right now. I'll check if that keeps working in the past it was also quite tricky to get it working (even toggling the option multiple times was not possible). 

I had valet installed previously and removed it before my initial installation. Could this be a problem and the reason why the php.ini might not be loaded correctly?

I've also deactivated the lines since I am using asdf for version management.
```
# Herd injected NVM configuration
# export NVM_DIR="/Users/andreasschneider/Library/Application Support/Herd/config/nvm"
# [ -s "$NVM_DIR/nvm.sh" ] && \. "$NVM_DIR/nvm.sh"  # This loads nvm
```

**Comment by blumewas** [comment]
I am going to close this issue since it seems to work again. I don't know why this happened.

**Comment by ArtDepartmentMJ** [comment]
I am also not able to get dumps working on 1.4.2

**Comment by mpociot** [comment]
Which error do you get? Can you try and toggle the dump interception?

**Comment by ArtDepartmentMJ** [comment]
No error, it just dumps it as normal (not in Herd). I have tried toggling a few times and it was eventually working for vanilla routes however it seems to have an issue, for me at least, when trying to dump via requests in Nova.

**Comment by sschlein** [comment]
@ArtDepartmentMJ The problem is related to a race condition, you can either toggle the dumps feature everytime when this happens or uncomment a few cards from your nova dashboard while working on one – that helped for me until we fixed it.

The fix will be released with Herd 1.5 next week.

**Comment by Jebble** [comment]
Arrived in this thread when my dumps were resulting in CORS errors in the browser.
Noticed that when I disabled CORS, dump would work but not be captured in herd.

It is fixed by toggling the intercept off and on again, I'm on Herd 1.6.2
